vignettes/functionality-flow.Rmd

```diff
@@ -0,0 +1,188 @@
+
+<!-- Q: How are we feeling about the colours? All internal functions could also just be light grey -->
+
+<!-- Q: Generally, Do we want to include arguments in inclusion and exclusion functions so a (technically inclined) user will be able to alter the algorithm for their needs? E.g., age could be an argument in `exclude_potential_pcos` -->
+
+### Inclusion event: HbA1c tests above 48 mmol/mol
+
+The function `include_hba1c_tests_over_48_mmol_per_mol` uses RLRR as
+data source to extract all events of tests above 48 mmol/mol before the
+index date.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Is it really *all* or is there a time window before the index date we're extracting from? -->
+
+### Inclusion event: Hospital diagnosis of diabetes
+
+The function `include_diabetes_diagnoses` uses the hospital contacts
+from DNPR to include all dates of diabetes diagnoses before the index
+date. Diabetes diagnoses from both ICD 8 and ICD 10 are included.
+
+This function contains two helper functions:
+
+- `extract_diabetes_diagnoses_icd_10`
+- `extract_diabetes_diagnoses_icd_8`
+
+<!-- TODO: Add details on how this filtering should be done, e.g., diagnosis codes -->
+
+<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+
+<!-- TODO: Is it really *all* or is there a time window before the index date we're extracting from? -->
+
+### Inclusion event: Diabetes-specific podiatrist services
+
+The function `include_diabetes_specific_podiatrist_services` uses the
+Health Service Register to extract the dates of all diabetes-specific
+podiatrist services before the index date.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Is it really *all* or is there a time window before the index date we're extracting from? -->
+
+### Inclusion event: GLD purchases
+
+The function `include_gld_purchases` uses Lægemiddelsdatabasen to
+extract the dates of all GLD purchases before the index date.
+
+<!-- TODO: Add English translation of data source -->
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+<!-- TODO: Is it really *all* or is there a time window before the index date we're extracting from? -->
+
+### Exclusion: HbA1c tests and GLD purchases during estimated pregnancy windows {#exclusion-pregnancy-windows}
+
+The function `exclude_events_during_pregnancy_windows` uses diagnoses
+from DNPR as a data source and is used to exclude both HbA1c tests and
+GLD purchases during estimated pregnancy windows.
+
+Under the hood, this relies on the function
+`extract_pregnancy_index_dates` and contains the following three helper
+functions:
+
+- `calculate_pregnancy_index_date_for_mc_visits_wo_end_date` (this
+  might be removed with the inclusion of )
+- `extract_pregnancy_end_dates`
+- `extract_maternal_care_visit_dates_without_end_date`
+
+`calculate_pregnancy_index_date_for_mc_visits_wo_end_date` might be
+removed with the inclusion of the birth register.
```

Comment:
Yes.

To add to this: The Medical Birth Register is much cleaner than the diagnosis data in the National Patient Register. The Birth Register only contains one row for each pregnancy with a variable containing the date of the end of the pregnancy.
The National Patient Register contains multiple rows for each pregnancy. Some contain diagnoses indicating the end of the pregnancy, similar to the Medical Birth Register, and some indicate maternal care visits.

The current implementation in the diagnosis data from the Patient Register is probably overkill. I'll test in the survey validation data if using only the dates of end-of-pregnancy diagnoses is enough to censor for GDM. I suspect that adding maternal care visits don't add any value in practice, and only complicates the code and the data necessary to run the algorithm.

I've added #74 to look more into this.

Reply:
Ah, nice. I'm looking forward to hearing about the results of those tests 🤞